zquery/shared/src/main/scala/zio/zquery/DataSource.scala

```diff
@@ -34,7 +37,7 @@ trait DataSource[-R, -A] { self =>
    * collection will contain at least one request and that all requests will be
    * unique when they are called by `ZQuery`.
    */
-  def run(requests: Iterable[A]): ZIO[R, Nothing, CompletedRequestMap]
+  def runAll(requests: Iterable[Iterable[A]]): ZIO[R, Nothing, CompletedRequestMap]
```

Comment:
`Iterable[Iterable[A]]` is not super clear, may be worth a Scaladoc note and / or newtype.

Reply:
Yes. We could potentially do something like:

```scala
final case class Sequential[+A](sequential: Iterable[A])
final case class Parallel[+A](parallel: Iterable[A])

requests: Sequential[Parallel[A]]
```

Though that does create a little more work for the user in digging into these types. Could also use type aliases though that adds a bit of indirection. At the very least agree we should add more documentation about this.

Reply:
Yes. I think the version of `Chunk` in the last release doesn't have quite enough combinators to switch over to in the signature of `runAll` yet but once we have the next version where `Chunk` extended `IndexedSeq` or is convertible into it I think we should do that.

Reply:
Sounds good!